### PR TITLE
AM-2841: Add jurisdiction filter to system user hearings role (sscs & private law)

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-2841

### Change description ###
Add jurisdiction filter to system user hearings role (sscs & private law)

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java [hunk 1]
```diff
@@ -627,6 +627,38 @@ void shouldDeleteOrgRequestedRoleForHearing(String roleName, String roleCategory
     }
 
 
+    @ParameterizedTest
+    @CsvSource({
+        "hearing-manager,SYSTEM,PRIVATELAW,Benefit",
+        "hearing-viewer,SYSTEM,PRIVATELAW,Benefit"
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java:633]: For read-ability the test Jurisdiction and Case Type should probably match up: e.g. 
* PRIVATELAW => PRLAPPS
* SSCS => Benefit
i.e. the only reason in the test to expect failure should be the mis-matched process and jurisdiction: not the mis-matched case-type  (I know this is not a rule enforced here but it is in many other rule/config scenarios).

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java [hunk 2]
```diff
@@ -691,4 +723,36 @@ void shouldDeletePrivateLawOrgRequestedRoleForHearing(String roleName,
             assertEquals(jurisdiction, roleAssignment.getAttributes().get("jurisdiction").asText());
         });
     }
+
+    @ParameterizedTest
+    @CsvSource({
+        "hearing-manager,SYSTEM,SSCS,PRLAPPS",
+        "hearing-viewer,SYSTEM,SSCS,PRLAPPS"
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java:730]: For read-ability the test Jurisdiction and Case Type should probably match up: e.g. 
* PRIVATELAW => PRLAPPS
* SSCS => Benefit
i.e. the only reason in the test to expect failure should be the mis-matched process and jurisdiction: not the mis-matched case-type  (I know this is not a rule enforced here but it is in many other rule/config scenarios).

**Review by mattnayler — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java (hunk 1 above)
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java]: Resolved in 14c0ad52b5d834d1ff2b0bc8584bdb5ad5cef610

File: src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java (hunk 2 above)
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/AllServicesOrgRoleTest.java]: Resolved in 14c0ad52b5d834d1ff2b0bc8584bdb5ad5cef610

**Review by mattnayler — APPROVED** [review]

**Review by abul-syed — APPROVED** [review]

**Pull request merged**